app/request_handler.py

```diff
@@ -46,6 +46,14 @@ def writeYelpRecords(yelpVenues):
 
     db.child(venuesTable).update(record)
 
+def writeVenueProviderRecord(yelpID, details, idObj = None):
+    try:
+        venue = representation.updateRecord(yelpID, **details)
+        for provider, data in venue["providers"].items():
+            db.child(venuesTable).child("details").child(yelpID).child("providers").update({provider: data})
```

Comment:
fyi: Not sure if it's intentional (if so maybe add comments?) but I've found it's significantly slower to update items in a collection individually rather than do batch updates (same with get operations).

I saw you get the full child, update the returned dictionary, and set with that – maybe use that trick again. :)